.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+name: Create a release on tag
+on: 
+  push: 
+    tags:
+      - "v*.*.*"
+jobs:
+  build: 
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+          cache: 'npm'
+
+      - name: Install Node dependencies
+        run: npm ci
+
+      - name: Set package version
+        run: |
+            LATEST_TAG=$(echo $GITHUB_REF | sed s/^v// | tr -d '"')
+            VERSION_NUMBER=".version=\"${LATEST_TAG}\""
+            jq ${VERSION_NUMBER} package.json > temp_package.json
+            mv temp_package.json package.json
```

Comment:
One issue with doing this only in CI is that if people build Kinto Admin the "old" way to customize build variables (that is, `curl` the source code then run `npm run build`), the version won't be overwritten.

What if we handled this in `vite.config.ts` instead? Here's a quick example that only gets the latest tag, but perhaps we could write a function that encompases all of the logic currently in the action workflow:

```diff
diff --git a/vite.config.ts b/vite.config.ts
index 9817bd7..f2d2111 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -2,6 +2,8 @@
 
 import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
+import { execSync } from 'child_process'
+
 /**
  * used to set the relative path from which we expect to serve the admin's
  * static bundle on the server:
@@ -9,7 +11,7 @@ import react from '@vitejs/plugin-react'
  *    Kinto plugin: /v1/admin/
  */
 const ASSET_PATH = process.env.ASSET_PATH || "/";
-
+const KINTO_ADMIN_VERSION = execSync('git describe --tags --abbrev=0').toString()
 // https://vitejs.dev/config/
 export default defineConfig({
   server: {
@@ -19,7 +21,7 @@ export default defineConfig({
   plugins: [react()],
   base: ASSET_PATH,  
   define: {
-    KINTO_ADMIN_VERSION: JSON.stringify(process.env.npm_package_version),
+    KINTO_ADMIN_VERSION: JSON.stringify(KINTO_ADMIN_VERSION)
   },
   build: {
     outDir: "build"

```

Reply:
I do like that this simplifies things. I'll combine this with @leplatrem 's suggestion above to do a `--abbrev=4`. If they're exactly on a tag, it'll just be the tag name returned. If they're beyond that tag, it'll  note how far ahead they are and which sha they're on.

Reply:
Implemented!